<launch>
  <!-- Declare command-line arguments with default values -->
  <arg name="audio_samplerate" default="48000" />
  <arg name="buffer" default="3072" />
  <arg name="device_index" default="6" />
  <arg name="audio_stream_topic" default="/audio_frames" />

  <!-- Set parameters from arguments -->
  <param name="audio_samplerate" value="$(arg audio_samplerate)" />
  <param name="buffer" value="$(arg buffer)" />
  <param name="device_index" value="$(arg device_index)" />
  <param name="audio_stream_topic" value="$(arg audio_stream_topic)" />

  <!-- Launch the microphone audio publisher node -->
  <node
    name="microphone_publisher"
    pkg="tiago_audio_streamer"
    type="microphone_publisher.py"
    output="screen"
    required="true"
  />
</launch>
